tests/BasicSearchTests.FunctionalTests.Core/V2SearchFunctionalTests.cs

```diff
@@ -37,5 +38,60 @@ public class V2SearchFunctionalTests : BaseFunctionalTests
             Assert.True(result.TotalHits.HasValue && result.TotalHits.Value > 0, "No results found, should find atleast some results for empty string query.");
             Assert.NotNull(result.Data);
         }
+
+        [Fact]
+        public void ShouldThrowExceptionForInvalidSearchResults()
```

Comment:
These are kind of unit tests on the `V2SearchResult` class, not functional tests. Do we have something that actually makes an HTTP request and asserts HTTP 500?